docs/yarn_classic.md

```diff
@@ -0,0 +1,147 @@
+   Corepack from showing the URL when it needs to download software;
+ * `COREPACK_ENABLE_PROJECT_SPEC` is set to "0" which prevents
+   Corepack from checking if the package manager corresponds to the one
+   defined for the current project;
+ * `YARN_IGNORE_PATH` is set to "true" which ignores any Yarn version specified by a user and
+   uses Corepack's version instead;
+ * `YARN_IGNORE_SCRIPTS`: is set to "true" which prevents execution of any scripts defined in
+   `package.json` or in any dependency;
+ * `YARN_YARN_OFFLINE_MIRROR` is set to point to `deps/yarn-classic` which is relative to
+   output directory and will hold fetched dependencies;
+ * `YARN_YARN_OFFLINE_MIRROR_PRUNING` is set to "false" which prevents Yarn from attempting to
+   ensure that dependencies are up to date.
+
+Once fetch phase is completed Cachi2 will generate an environment file with variables pointing
+to the mirror and instructing Yarn not to prune it:
+
+```
+YARN_YARN_OFFLINE_MIRROR=<request output directory>
+YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+```
+
+Sourcing this file will prime Yarn for an offline build
+
+Cachi2 expects to find well-formed `package.json` and `yarn.lock` checked in into a
+repository and will not continue if any of the files are missing.
+
+Prefetching dependencies for Yarn in Cachi2 is be done using Yarn's offline mirror feature.
+The project must be configured to use the offline mirror feature. Refer to
+[Setting up offline mirror](#setting-up-offline-mirror) below for details.
+
+
+### Supported dependencies types
+
+Yarn Classic PM is capable of processing the following types of dependencies:
+ * packages from registries;
+ * packages from git repos;
+ * packages from http/https URLs;
+ * packages from local paths;
+ * local workspace packages;
+ * local link packages.
+
+### Dealing with Yarn Zero-Installs
+
+Yarn's Classic [Plug'n'Play](https://classic.yarnpkg.com/en/docs/pnp) feature is not
+supported. Any package that uses it will be rejected. For further details
+please refer to [Yarn v3 documentation](yarn.md).
+
+## Fetching dependencies for Yarn Classic projects
+
+The process of fetching dependencies for Yarn Classic is similar to that for any other
+package manager. The name of package manager is `yarn_classic`, and it does not expect
+any additional arguments.
+
+Cachi2 ``fetch-deps`` shell command:
+
+```shell
+cachi2 fetch-deps \
+  --source ./my-repo \
+  --output ./cachi2-output \
+  '<JSON input>'
+```
+
+where JSON input is:
+```jsonc
+{
+  // "yarn_classic" tells Cachi2 to process Yarn packages
+  "type": "yarn_classic",
+  // path to the package (relative to the --source directory)
+  // defaults to "."
+  "path": ".",
+}
+```
+
+or more simply by just invoking:
+``cachi2 fetch-deps yarn_classic``
+
+For complete example of how to pre-fetch dependencies, see [Pre-fetch dependencies][usage-prefetch].
+
+## Controlling Yarn's behavior with yarn-specific config files during prefetch
```

Comment:
This section is a bit unclear/confusing.

We start off talking about "Controlling Yarn's {prefetch} behavior", but then immediately state that users *can't* control Yarn's behavior during prefetch.
Then we go on to talk about the build phase...

Reply:
Good point, thank you. Renamed it.